### DISCUSSION TITLE
Two feature suggestions: Field "Instructions" and "slug" field

### DISCUSSION THREAD

**Discussion by schwartzmj** [pocketbase/pocketbase#3241] [Q&A] [answered]
Has there been any interest in these two features being added to Pocketbase?

- When creating a field on a collection, add support for a "description" or "instructions" for the field. This allows us to elaborate on what a field is for and specifically what it does for the end user. It also allows us to provide an example. Imagine a field called "phone" with a description of "The staff member's phone number to be displayed within their profile card in the format of xxx-xxx-xxxx". This is more helpful than the user entering random values until they figure out how to satisfy the regex and also provides more detail on how it is used.
- Implement a new "slug" field that auto-generates a slug based on another field, e.g. `title` or maybe even multiple other fields. It would need the ability to be overridden/editable. Maybe some other feature could accomplish this like configurable "auto-populated" fields.

As it stands, we find it difficult to use Pocketbase for non-tech savvy users and believe these two features would go a long way towards helping with that.

**Comment by ganigeorgiev** [comment, marked as answer]
I've added the field "instructions"/comments suggestion to the [roadmap](https://github.com/orgs/pocketbase/projects/2/views/1?pane=issue&itemId=37392385). I'm still reluctant to implement it as it may clutter the UI but I'll reconsider it since several other users also requested it in the past. 

The `slug` field could complicate the internal validations as it will be dependent on other fields and at least for v1 remains out of the scope (_the pile of feature requests doesn't end and I have to put a stop somewhere as I'm already almost 6months behind the roadmap of my other open source project..._).

Note that while the Admin UI can be used as a very rudimentary CMS, it wasn't really intended for that as it is lacking a lot of features that non-tech savvy users will find missing.
If you are looking for a more CMS-like experience you might find [Strapi](https://strapi.io/) or [Directus](https://directus.io/) better for your use case.

**Reply by schwartzmj** [reply]
Appreciate the response! Happy to hear you're considering the "instructions" field as we'd find that extremely helpful.

Pocketbase is fantastic, thank you for your work!

**Comment by schwartzmj** [comment]
> The `slug` field could complicate the internal validations as it will be dependent on other fields and at least for v1 remains out of the scope (_the pile of feature requests doesn't end and I have to put a stop somewhere as I'm already almost 6months behind the roadmap of my other open source project..._).

One other thought here. I envisioned the `slug` field really being more of a front-end "helper" field. It would really just be a text field, but would auto-complete client-side as the user types in a "title" (or clicks a "generate" button) and would have a unique constraint by default. Totally understand if this doesn't make sense for your vision though!

**Comment by einarpersson** [comment]
I am not sure that an `instruction` for each field is such a great idea. There are so many fields where this is not relevant, where the values are not inserted manually.

_If_ it were to be implemented I think I would prefer the name `description` as that could be applicable to all fields, and that could still bring some value and clarity. But I am not sure. Anyhow, given our usage of Pocketbase, I think that I as a new user would get confused if I saw `instruction` (but that could be solved with an `instruction` for that 🤣 ). Description is more descriptive.

**Reply by schwartzmj** [reply]
Agree, "description" makes sense for a naming convention.

**Comment by jnsprnw** [comment]
I would also like to suggest adding a placeholder option for each field. While field descriptions are sometimes preferable, a simple placeholder can also help the user.
I often set up Pocketbase for non-technical clients, family and friends to collaborate on content input. Descriptions and placeholders would be very beneficial for that purpose.
